pkgs/top-level/all-packages.nix

```diff
@@ -21670,11 +21670,6 @@ in
     patches = config.dwm.patches or [];
```

Comment:
Is this actually missing `conf = config.dwm.conf or null;` that `dwm-git` has?
How does `conf ? null` get populated otherwise...

Should that be added in this PR too or another PR?

@AndersonTorres

Reply:
How many packages still use the config interface? I think it is 'reserved' for NixOS or external programs like Home Manager.

User-provided patches are usually a parameter with null as default.

Reply:
:shrug: I have no idea

I just saw `conf` was on the `dwm-git` copy but not on `dwm` but they both include it on the first line.

Reply:
Get rid of that config reference.

Reply:
The patches = config.dwm.patches part?
Do i also need to change the nixos module?

Reply:
The `patches = config.dwm.patches` reference.

Is the module change needed?

Reply:
I don't know, thats why I'm asking :sweat_smile:
I dont have much experience with nixos modules

Reply:
I think we can keep it here for backwards compatibility. Long-term we hopefully get a better nix interface for expressing packages overrides.

Reply:
I have read the module, and it has no reference about configuring patches. Then it needs no modification at all.

Reply:
Ok I've removed it. LMK if that's right

Reply:
It might break people still that used this option before.

Reply:
I guess if someone runs into the issue we add it back in?
Or do we direct them to do it a different way such as with overrides?

Reply:
I don't understand why the config reference was removed, I use nixpkgs.config.dwm.patches to set my patches and this needlessly broke my config on 21.05 upgrade. I'm making an issue to add this feature back.